File: src/main/java/com/jss/osiris/modules/quotation/service/NoticeTypeService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.NoticeType;

public interface NoticeTypeService {
    public List<NoticeType> getNoticeTypes();

    public NoticeType getNoticeType(Integer id);
}

File: client-myjss/src/app/modules/quotation/services/notice.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../main/services/appRest.service';
import { NoticeType } from '../model/NoticeType';

@Injectable({
  providedIn: 'root'
})
export class NoticeTypeService extends AppRestService<NoticeType> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getNoticeTypes() {
    return this.getListCached(new HttpParams(), "notice-types");
  }
}
